config/proposer/loader.go

```diff
@@ -0,0 +1,281 @@
+			}
+			loadConfig = psl.processProposerSettings(settingFromFile, loadConfig)
+		case urlFlag:
+			var settingFromURL *validatorpb.ProposerSettingsPayload
+			if err := config.UnmarshalFromURL(cliCtx.Context, cliCtx.String(flags.ProposerSettingsURLFlag.Name), &settingFromURL); err != nil {
+				return nil, err
+			}
+			if settingFromURL == nil {
+				return nil, errors.New("proposer settings is empty after unmarshalling from url")
+			}
+			loadConfig = psl.processProposerSettings(settingFromURL, loadConfig)
+		case onlyDB:
+			loadConfig = psl.processProposerSettings(nil, loadConfig)
+		case none:
+			if psl.options.builderConfig != nil {
+				// if there are no proposer settings provided, create a default where fee recipient is not populated, this will be skipped for validator registration on validators that don't have a fee recipient set.
+				// skip saving to DB if only builder settings are provided until a trigger like keymanager API updates with fee recipient values
+				option := &validatorService.ProposerOption{
+					BuilderConfig: psl.options.builderConfig.Clone(),
+				}
+				if loadConfig == nil {
+					loadConfig = &validatorpb.ProposerSettingsPayload{}
+				}
+				loadConfig.DefaultConfig = option.ToConsensus()
+			}
+		default:
+			return nil, errors.New("load method for proposer settings does not exist")
+		}
+	}
+
+	// exit early if nothing is provided
+	if loadConfig == nil {
+		log.Warn("No proposer settings were provided")
+		return nil, nil
+	}
+	ps, err := validatorService.ProposerSettingFromConsensus(loadConfig)
+	if err != nil {
+		return nil, err
+	}
+	if err := psl.db.SaveProposerSettings(cliCtx.Context, ps); err != nil {
+		return nil, err
+	}
+	return ps, nil
+}
+
+func (psl *SettingsLoader) processProposerSettings(loadedSettings, dbSettings *validatorpb.ProposerSettingsPayload) *validatorpb.ProposerSettingsPayload {
+	dbOnly := false
+	if loadedSettings == nil && dbSettings == nil {
+		return nil
+	}
+	// fill in missing data from db
+	if loadedSettings == nil && dbSettings != nil {
+		dbOnly = true
+		loadedSettings = dbSettings
+	}
+	if loadedSettings.DefaultConfig == nil && dbSettings != nil && dbSettings.DefaultConfig != nil {
+		loadedSettings.DefaultConfig = dbSettings.DefaultConfig
+	}
+	if loadedSettings.ProposerConfig == nil && dbSettings != nil && dbSettings.ProposerConfig != nil {
+		loadedSettings.ProposerConfig = dbSettings.ProposerConfig
+	}
+	// if default and proposer configs are both missing even after db setting
+	if loadedSettings.DefaultConfig == nil && loadedSettings.ProposerConfig == nil {
+		return nil
+	}
+
+	if loadedSettings.DefaultConfig != nil {
+		if loadedSettings.DefaultConfig.Builder != nil {
+			loadedSettings.DefaultConfig.Builder.GasLimit = reviewGasLimit(loadedSettings.DefaultConfig.Builder.GasLimit)
+		}
+		// override the db settings with the results based on whether the --enable-builder flag is provided.
```

Comment:
This overrides the `loadedSettings`, not the `dbSettings`.

Reply:
will update though in this case the loadedSettings == db settings